src/csharp/Microsoft.Spark/Broadcast.cs

```diff
@@ -0,0 +1,284 @@
+
+            if (encryptionEnabled)
+            {
+                throw new NotImplementedException("Broadcast encryption is not supported yet.");
+            }
+            else
+            {
+                WriteToFile(value);
+            }
+
+            var pythonBroadcast = (JvmObjectReference)javaSparkContext.Jvm.CallStaticJavaMethod(
+                "org.apache.spark.api.python.PythonRDD",
+                "setupBroadcast",
+                _path);
+
+            return (JvmObjectReference)javaSparkContext.Jvm.CallNonStaticJavaMethod(
+                javaSparkContext,
+                "broadcast",
+                pythonBroadcast);
+        }
+
+        /// <summary>
+        /// Function that creates a file in _path to store the broadcast value in the given path.
+        /// </summary>
+        /// <param name="value">Broadcast value to be written to the file</param>
+        private void WriteToFile(object value)
+        {
+            using FileStream f = File.Create(_path);
+            Dump(value, f);
+        }
+
+        /// <summary>
+        /// Function that serializes and stores the object passed to the given Stream.
+        /// </summary>
+        /// <param name="value">Serializable object</param>
+        /// <param name="stream">Stream to which the object is serialized</param>
+        private void Dump(object value, Stream stream)
+        {
+            var formatter = new BinaryFormatter();
+            formatter.Serialize(stream, value);
+        }
+    }
+
+    /// <summary>
+    /// Global registry to store the object value of all active broadcast variables from
+    /// the workers. This registry is only used on the worker side when Broadcast.Value() is called
+    /// through a UDF.
+    /// </summary>
+    internal static class BroadcastRegistry
+    {
+        private static readonly ConcurrentDictionary<long, object> s_registry =
+            new ConcurrentDictionary<long, object>();
+
+        /// <summary>
+        /// Function to add the value of the broadcast variable to s_registry.
+        /// </summary>
+        /// <param name="bid">Id of the Broadcast variable object to add</param>
+        /// <param name="value">Value of the Broadcast variable</param>
+        internal static void Add(long bid, object value)
+        {
+            bool result = s_registry.TryAdd(bid, value);
+            if (!result)
+            {
+                Console.WriteLine($"Broadcast {bid} already exists in the registry.");
+            }
+        }
+
+        /// <summary>
+        /// Function to remove the Broadcast variable from s_registry.
+        /// </summary>
+        /// <param name="bid">Id of the Broadcast variable object to remove</param>
+        internal static void Remove(long bid)
+        {
+            if (!s_registry.TryRemove(bid, out _))
+            {
+                throw new Exception($"Trying to remove a broadcast {bid} that does not exist.");
```

Comment:
Do you think logging this at warning level is fine?

Reply:
Yes that makes sense, will do that, thanks!